test_vm/tests/commit_post_test.rs

```diff
@@ -139,22 +156,268 @@ fn commit_post_flow_happy_path() {
+                        subinvocs: Some(vec![ExpectInvocation {
+                            to: *STORAGE_POWER_ACTOR_ADDR,
+                            method: PowerMethod::EnrollCronEvent as u64,
+                            ..Default::default()
+                        }]),
+                        ..Default::default()
+                    },
+                    ExpectInvocation {
+                        to: *REWARD_ACTOR_ADDR,
+                        method: RewardMethod::UpdateNetworkKPI as u64,
+                        ..Default::default()
+                    },
+                ]),
+                ..Default::default()
+            },
+            ExpectInvocation {
+                to: *STORAGE_MARKET_ACTOR_ADDR,
+                method: MarketMethod::CronTick as u64,
+                ..Default::default()
+            },
+        ]),
+        ..Default::default()
+    }
+    .matches(v.take_invocations().last().unwrap());
+
+    // power unproven so network stats are the same
+    let network_stats = v.get_network_stats();
+    assert_eq!(StoragePower::zero(), network_stats.total_bytes_committed);
+    assert!(network_stats.total_pledge_collateral.is_positive());
+}
+
+#[test]
+fn overdue_precommit() {
+    let store = MemoryBlockstore::new();
+    let policy = &Policy::default();
+    let mut v = VM::new_with_singletons(&store);
+    let addrs = create_accounts(&v, 1, TokenAmount::from(10_000e18 as i128));
+    let seal_proof = RegisteredSealProof::StackedDRG32GiBV1P1;
+    let (owner, worker) = (addrs[0], addrs[0]);
+    let id_addr = create_miner(
+        &mut v,
+        owner,
+        worker,
+        seal_proof.registered_window_post_proof().unwrap(),
+        TokenAmount::from(10_000e18 as i128),
+    )
+    .0;
+    let mut v = v.with_epoch(200);
+
+    // precommit and advance to prove commit time
+    let sector_number: SectorNumber = 100;
+    let precommit =
+        precommit_sectors(&mut v, 1, 1, worker, id_addr, seal_proof, sector_number, true, None)
+            .get(0)
+            .unwrap()
+            .clone();
+
+    let balances = v.get_miner_balance(id_addr);
+    assert!(balances.pre_commit_deposit.is_positive());
+
+    let prove_time = v.get_epoch() + max_prove_commit_duration(policy, seal_proof).unwrap() + 1;
+    v = advance_by_deadline_to_epoch(v, id_addr, prove_time).0;
+
+    //
+    // overdue precommit
+    //
+
+    // advance time to precommit clean up epoch
+    let cleanup_time = prove_time + policy.expired_pre_commit_clean_up_delay;
+    let (mut v, deadline_info) = advance_by_deadline_to_epoch(v, id_addr, cleanup_time);
+
+    // advance one more deadline so precommit clean up is reached
+    v = v.with_epoch(deadline_info.close);
+
+    // run cron which should clean up precommit
+    apply_ok(
+        &v,
+        *SYSTEM_ACTOR_ADDR,
+        *CRON_ACTOR_ADDR,
+        TokenAmount::zero(),
+        CronMethod::EpochTick as u64,
+        RawBytes::default(),
+    );
+
+    ExpectInvocation {
```

Comment:
could we contrive a nicer, perhaps fluent API for expected invocations? Right now it feels a bit rough.

Reply:
Yeah, I think some cleanup is needed of this, but it's gonna be deferred to a later issue.

Reply:
@LesnyRumcajs if you have any ideas on improving they would be valuable to preserve please write them up in an issue or feel free to throw them out on slack and I can gather them together